Notesnook.API/Controllers/InboxController.cs

```diff
@@ -115,5 +127,58 @@ public async Task<IActionResult> DeleteApiKeyAsync(string apiKey)
                 return BadRequest(new { error = ex.Message });
             }
         }
+
+        [HttpGet("public-encryption-key")]
+        [Authorize(Policy = InboxApiKeyAuthenticationDefaults.AuthenticationScheme)]
+        public async Task<IActionResult> GetPublicKeyAsync()
+        {
+            var userId = User.FindFirstValue("sub");
+            try
+            {
+                var userSetting = await UserSetting.FindOneAsync(u => u.UserId == userId);
+                if (string.IsNullOrWhiteSpace(userSetting?.InboxKeys?.Public))
+                {
+                    return BadRequest(new { error = "Inbox public key is not configured." });
+                }
+                return Ok(new { key = userSetting.InboxKeys.Public });
+            }
+            catch (Exception ex)
+            {
+                await Slogger<InboxController>.Error(nameof(GetPublicKeyAsync), "Couldn't get user's inbox's public key.", userId, ex.ToString());
+                return BadRequest(new { error = ex.Message });
+            }
+        }
+
+        [HttpPost("items")]
+        [Authorize(Policy = InboxApiKeyAuthenticationDefaults.AuthenticationScheme)]
+        public async Task<IActionResult> CreateInboxItemAsync([FromBody] SyncItem request)
+        {
+            var userId = User.FindFirstValue("sub");
+            try
+            {
+                if (request.Algorithm != "xcha-argon2i13-7")
+                {
+                    return BadRequest(new { error = "Invalid algorithm. Only xcha-argon2i13-7 is supported." });
+                }
+                if (string.IsNullOrWhiteSpace(request.Cipher) || string.IsNullOrWhiteSpace(request.IV))
+                {
+                    return BadRequest(new { error = "Inbox item cipher and iv are required." });
+                }
+                if (request.Length <= 0)
+                {
+                    return BadRequest(new { error = "Valid inbox item length is required." });
+                }
+
+                request.UserId = userId;
+                request.Version = 6.1;
+                await InboxItems.InsertAsync(request);
+                return Ok(new { message = "Inbox item created successfully." });
+            }
```

Comment:
I'm anticipating some of this implementation to change while working on the inbox api server